tests/components/enphase_envoy/test_config_flow.py

```diff
@@ -397,6 +399,186 @@ async def test_zeroconf_host_already_exists(
+
+
+async def test_zero_conf_malformed_serial_property(
+    hass: HomeAssistant, config_entry, setup_enphase_envoy
+) -> None:
+    """Test malformed zeroconf properties."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+
+    with pytest.raises(Exception):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_ZEROCONF},
+            data=zeroconf.ZeroconfServiceInfo(
+                ip_address=ip_address("1.1.1.1"),
+                ip_addresses=[ip_address("1.1.1.1")],
+                hostname="mock_hostname",
+                name="mock_name",
+                port=None,
+                properties={"serilnum": "1234", "protovers": "7.1.2"},
+                type="mock_type",
+            ),
+        )
+        await hass.async_block_till_done()
+
+
+async def test_zero_conf_malformed_serial(
+    hass: HomeAssistant, config_entry, setup_enphase_envoy
+) -> None:
+    """Test malformed zeroconf properties."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_ZEROCONF},
+        data=zeroconf.ZeroconfServiceInfo(
+            ip_address=ip_address("1.1.1.1"),
+            ip_addresses=[ip_address("1.1.1.1")],
+            hostname="mock_hostname",
+            name="mock_name",
+            port=None,
+            properties={"serialnum": "12%4", "protovers": "7.1.2"},
+            type="mock_type",
+        ),
+    )
+    await hass.async_block_till_done()
+    assert result["type"] is FlowResultType.FORM
+    assert config_entry.data["host"] == "1.1.1.1"
+    assert config_entry.unique_id == "1234"
+    assert config_entry.title == "Envoy 1234"
+
+
+async def test_zero_conf_malformed_fw_property(
+    hass: HomeAssistant, config_entry, setup_enphase_envoy
+) -> None:
+    """Test malformed zeroconf property."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_ZEROCONF},
+        data=zeroconf.ZeroconfServiceInfo(
+            ip_address=ip_address("1.1.1.1"),
+            ip_addresses=[ip_address("1.1.1.1")],
+            hostname="mock_hostname",
+            name="mock_name",
+            port=None,
+            properties={"serialnum": "1234", "protvers": "7.1.2"},
+            type="mock_type",
+        ),
+    )
+    await hass.async_block_till_done()
+    assert result["type"] is FlowResultType.ABORT
```

Comment:
We should always assert the reason when the type is abort

Reply:
done